lib/spack/spack/test/compilers/basics.py

```diff
@@ -1,940 +0,0 @@
-# Copyright Spack Project Developers. See COPYRIGHT file for details.
-#
-# SPDX-License-Identifier: (Apache-2.0 OR MIT)
-"""Test basic behavior of compilers in Spack"""
-import json
-import os
-from copy import copy
-from typing import Optional
-
-import pytest
-
-import llnl.util.filesystem as fs
-
-import spack.compiler
-import spack.compilers
-import spack.config
-import spack.spec
-import spack.util.module_cmd
-from spack.compiler import Compiler
-from spack.util.executable import Executable, ProcessError
-from spack.util.file_cache import FileCache
-
-
-def test_multiple_conflicting_compiler_definitions(mutable_config):
-    compiler_def = {
-        "compiler": {
-            "flags": {},
-            "modules": [],
-            "paths": {"cc": "cc", "cxx": "cxx", "f77": "null", "fc": "null"},
-            "extra_rpaths": [],
-            "operating_system": "test",
-            "target": "test",
-            "environment": {},
-            "spec": "clang@0.0.0",
-        }
-    }
-
-    compiler_config = [compiler_def, compiler_def]
-    compiler_config[0]["compiler"]["paths"]["f77"] = "f77"
-    mutable_config.update_config("compilers", compiler_config)
-
-    arch_spec = spack.spec.ArchSpec(("test", "test", "test"))
-    cmp = spack.compilers.compiler_for_spec("clang@=0.0.0", arch_spec)
-    assert cmp.f77 == "f77"
-
-
-def test_compiler_flags_from_config_are_grouped():
-    compiler_entry = {
-        "spec": "intel@17.0.2",
-        "operating_system": "foo-os",
-        "paths": {"cc": "cc-path", "cxx": "cxx-path", "fc": None, "f77": None},
-        "flags": {"cflags": "-O0 -foo-flag foo-val"},
-        "modules": None,
-    }
-
-    compiler = spack.compilers.compiler_from_dict(compiler_entry)
-    assert any(x == "-foo-flag foo-val" for x in compiler.flags["cflags"])
```

Comment:
should this be translated for the new `packages.yaml` format?

Reply:
I don't think so. It should be covered by:
```
lib/spack/spack/test/concretization/flag_mixing.py::test_flag_order_and_grouping
```